dropwizard-auth/src/main/java/io/dropwizard/auth/AuthDynamicFeature.java

```diff
@@ -21,13 +21,26 @@
  * <p>If authorization is not a concern, then {@link RolesAllowedDynamicFeature}
  * could be omitted. But to enable authentication, the {@link PermitAll} annotation
  * should be placed on the corresponding resource methods.</p>
+ * <p>Note that registration of the filter will be done following the semantics of
+ * {@link FeatureContext#register(Class)} and {@link FeatureContext#register(Object)}:
+ * passing the filter as a {@link Class} to the {@link #AuthDynamicFeature(Class)}
+ * constructor will do dependency injection while passing the filter as an object to
+ * the {@link #AuthDynamicFeature(ContainerRequestFilter)} constructor won't.</p>
```

Comment:
Nit: "passing the filter as a {@link Class} to the {@link #AuthDynamicFeature(Class)} constructor will result in dependency injection, while objects passed to {@link #AuthDynamicFeature(ContainerRequestFilter)} will be used directly."

Reply:
how to make it work.
I need to register a custom intance of requestFilter from dynamic feature and want to get all injected objects with @Context or @inject injected.
In my case, I need a custom instance of requestFilter because i override RolesAllowedDynamicFeature to support i18n response output. so instead to instanciate your RolesAllowedRequestFilter within AuthorizationDynamicFeature i need to inject a messageInterpolator that i have created and binded

Reply:
please replace instead by despite the fact of. Sorry for my bad english

Reply:
@meilleureVie please open a ticket or ask on the google group for help, this is a merged PR.